x-pack/plugins/stack_alerts/public/alert_types/geo_containment/query_builder/util_components/geo_index_pattern_select.tsx

```diff
@@ -41,6 +41,9 @@ export class GeoIndexPatternSelect extends Component<Props, State> {
 
   componentDidMount() {
     this._isMounted = true;
+    if (this.props.value) {
+      this._onIndexPatternSelect(this.props.value);
```

Comment:
What do you think about breaking _onIndexPatternSelect into 2 methods so mounting component does not call props.onChange?

```
_loadIndexPattern = async (indexPatternId: string) => {
  if (!indexPatternId || indexPatternId.length === 0 || !this.props.indexPatternService) {
    return;
  }

  let indexPattern;
  try {
    indexPattern = await this.props.indexPatternService.get(indexPatternId);
  } catch (err) {
    return;
  }

  if (!this._isMounted || indexPattern.id !== indexPatternId) {
    return;
  }

  this.setState({
    doesIndexPatternHaveGeoField: indexPattern.fields.some((field) => {
      return this.props.includedGeoTypes.includes(field.type);
    }),
  });

  return indexPattern;
};

_onIndexPatternSelect = async (indexPatternId: string) => {
  const indexPattern = await this._loadIndexPattern(indexPatternId);
  if (indexPattern) {
    this.props.onChange(indexPattern);
  }
};
```

Reply:
makes sense, can call `_loadIndexPattern` then.